### PR TITLE
Support alarm tags

### LINKED ISSUE #5124: Support alarm labels
Please answer these questions before submitting your issue.

- Why do you submit this issue?
- [x] Question or discussion
- [ ] Bug
- [ ] Requirement
- [x] Feature or performance improvement

___
### Question
- Alarm support level

Many monitoring system alarms support levels, such as info, warn, critical.
I want to enhance the alarm rules with alert level.

### PR DESCRIPTION
**Pull request by liqiangz**
### Support alarm tags
- [ ] If this is non-trivial feature, paste the links/URLs to the design doc.
- [x] Update the documentation to include this new feature.
- [x] Tests(including UT, IT, E2E) are added to verify the new feature.
- [ ] If it's UI related, attach the screenshots below.

- [x] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #5124 #6821 .
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/CHANGES.md).

Related  https://github.com/apache/skywalking-query-protocol/pull/56

### CODE REVIEW

**Comment by wu-sheng** [comment]
Should I expect e2e failing as no submodule update? And more importantly, GraphQL changes have not been approved.

**Comment by liqiangz** [comment]
Wait a moment, I will check the fail.

**Comment by wu-sheng** [comment]
> Wait a moment, I will check the fail.

You should not to update anything. As the submodule relating to the repo, you don't have the commit id to update.

**Comment by liqiangz** [comment]
Yes,  e2e failing as no submodule 
image

File: oap-server/server-alarm-plugin/src/main/proto/alarm-hook.proto [hunk 1]
```diff
@@ -35,6 +35,17 @@ message AlarmMessage {
...
+}
+
+message AlarmTags {
+    // String key, String value pair.
+    repeated KeyStringValuePair data = 1;
+}
+
+message KeyStringValuePair {
+    string key = 1;
+    string value = 2;
```
Comment (wu-sheng) [inline oap-server/server-alarm-plugin/src/main/proto/alarm-hook.proto:48]: No document update about this.

**Review by wu-sheng — COMMENTED** [review]
Please update related documentation of alarm.

**Comment by wu-sheng** [comment]
@kezhenxu94 Please check on the e2e.

@arugal @EvanLjp @dmsolr Please review on tag-search related codes.

File: .github/workflows/e2e.alarm.yaml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      - '!**.md'
+  schedule:
+    - cron: '0 18 * * *'
+
+env:
+  SW_AGENT_JDK_VERSION: 8
+
+jobs:
+  Alalrm:
+    name: Alalrm
```
Comment (wu-sheng) [inline .github/workflows/e2e.alarm.yaml:32]:
```suggestion
    name: Alarm
```

Typo.

**Review by kezhenxu94 — APPROVED** [review]
LGTM

**Review by arugal — DISMISSED** [review]
LGTM

**Comment by wu-sheng** [comment]
@liqiangz I will wait for the document update before merge.

**Review by EvanLjp — APPROVED** [review]
LGTM

**Comment by wu-sheng** [comment]
Thanks for updating. Let's wait for the CI. 

@Fine0830 Once this gets merged, I will update this on the demo env, you could work on #6821.

**Review by wu-sheng — APPROVED** [review]
Approved by @EvanLjp @kezhenxu94 @dmsolr 

I rechecked the doc update.

**Pull request merged**